packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -206,6 +206,76 @@ export const makeZoeHelpers = zcf => {
         });
         zoeService.offer(invite);
       }),
+    /**
+     * Escrow a payment with Zoe and reallocate the amount of the
+     * payment to a recipient.
+     *
+     * @param {Object} obj
+     * @param {Payment} obj.payment
+     * @param {String} obj.keyword
+     * @param {Handle} obj.recipientHandle
+     * @returns {Promise<undefined>}
+     *
+     */
+    escrowAndAllocateTo: ({ payment, keyword, recipientHandle }) => {
+      // We will create a temporary offer to be able to escrow our payment
+      // with Zoe.
+      let tempHandle;
+
+      const amountMath = zcf.getAmountMaths(harden([keyword]))[keyword];
+      const { issuerKeywordRecord } = zcf.getInstanceRecord();
+      const issuer = issuerKeywordRecord[keyword];
+
+      // We need to make an invite and store the offerHandle of that
+      // invite for future use.
+      const contractSelfInvite = zcf.makeInvitation(
+        offerHandle => (tempHandle = offerHandle),
+      );
+      // To escrow the payment, we must get the Zoe Service facet and
+      // make an offer
+      return E(issuer)
+        .getAmountOf(payment)
```

Comment:
I don't like doing a round trip to the issuer merely to get an amount that the caller already knew. I'd prefer that the caller pass it in as an additional argument, even though that makes the caller a bit more verbose.